src/plugins/jingle/chat-header-notification.js

```diff
@@ -0,0 +1,45 @@
+import { CustomElement } from 'shared/components/element.js';
+import { _converse, api, converse } from "@converse/headless/core";
+import tpl_header_button from "./templates/header-button.js";
+import { JINGLE_CALL_STATUS } from "./constants.js";
+
+import './styles/jingle.scss';
+const { Strophe, $msg } = converse.env;
+
+export default class CallNotification extends CustomElement {
+    
+    static get properties() {
+        return {
+            'jid': { type: String },
+        }
+    }
+
+    initialize() {
+        this.model = _converse.chatboxes.get(this.jid);
+        this.listenTo(this.model, 'change:jingle_status', () => this.requestUpdate());
+    }
+    
+    render() {
+        return tpl_header_button(this);
+    }
+
+    endCall() {
+        const jingle_status = this.model.get('jingle_status');
+        if ( jingle_status === JINGLE_CALL_STATUS.OUTGOING_PENDING || jingle_status === JINGLE_CALL_STATUS.ACTIVE) {
+            this.model.save('jingle_status', JINGLE_CALL_STATUS.ENDED);
+            const stanza = $msg({
+                'from': _converse.bare_jid,
+                'to': this.jid,
+                'type': 'chat'
+            }).c('retract', {'xmlns': Strophe.NS.JINGLEMESSAGE, 'id': this.getAttribute('id')})
```

Comment:
The `id` attribute of the `retract` element in this stanza must match the `id` attribute of the `propose` element that was originally sent.

Otherwise there is no way for a recipient of this stanza to know which proposed call it relates to (they might have received more than one proposed call from multiple senders).

Currently when you send the `propose` stanza, you're not saving its data anywhere for the sender.

So you need to make sure that a message object is created and stored for the sender when the `propose` stanza is sent out.

There are multiple potential ways in which this could potentially be done. For now, you can try to just call `model.messages.create(attrs)` (where `model` is the chatbox) when you send out the `propose` stanza. That might be the easiest to do for now.

Then, when you send out the `retract` stanza, you need to get the latest `propose` stanza (e.g. via `model.messages.find`) and then get the value of the `id` that you used for the `propose` element, and then use that on the `retract` element here.